### PR TITLE
Azalapa/fac 164/subscripttion thread join timeout

### PR DESCRIPTION
**Pull request by antoniozlp**
# Description

 - Execute callback function in a safe way (try/except) to avoid subscription thread crash.
 - change thread.isAlive() to thread.is_alive() for python 3.9+

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- `pytest`
Screenshot 2023-03-26 at 14 52 06

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pygqlc/GraphQLClient.py [hunk 1]
```diff
@@ -574,7 +578,7 @@ def resetSubsConnection(self):
     if not self.sub_router_thread:
       print('connection not stablished, nothing to reset')
       return False
-    if self.sub_router_thread.isAlive(): #check that _sub_routing_loop() is running
+    if self.sub_router_thread.is_alive(): #check that _sub_routing_loop() is running
```
Comment (barucAlmaguer) [inline pygqlc/GraphQLClient.py:581]: si se cambia para que funcione en 3.9, de una vez lo prepararía para jalar en cualquier version de python asi:
```python
def safe_is_alive(thread: Threading.Thread) -> bool:
    import sys
    if sys.version_info[:2] >= (3, 9):
        return thread.is_alive()
    else:
        return thread.isAlive()
```

Comment (barucAlmaguer) [inline pygqlc/GraphQLClient.py:581]:
```suggestion
    if safe_is_alive(self.sub_router_thread): #check that _sub_routing_loop() is running
```

Comment (barucAlmaguer) [reply pygqlc/GraphQLClient.py:581]: (para evitar breaking changes si esto le funcionaba a una implementacion debajo de 3.9)

**Review by barucAlmaguer — COMMENTED** [review]
LGTM, solo falta añadir la versión en __version__ y puse un mini comentario

File: pygqlc/GraphQLClient.py (hunk 1 above)
Comment (antoniozlp) [reply pygqlc/GraphQLClient.py:581]: No lo veo tan necesario @barucAlmaguer desde la versión de [2.6 de python](https://docs.python.org/2.7/library/threading.html#threading.Thread.isAlive) se agredo el `is_alive()`, los dos funcionaban de la misma manera hasta la en 3.9 quitaron el `isAlive()`

**Pull request merged**